youtube_dl/extractor/condenast.py

```diff
@@ -119,22 +121,37 @@ def _extract_series(self, url, webpage):
     def _extract_video_params(self, webpage):
         query = {}
         params = self._search_regex(
-            r'(?s)var params = {(.+?)}[;,]', webpage, 'player params', default=None)
+            r'(?s)\bparams = {(.+?)}[;,]', webpage, 'player params', default=None)
         if params:
             query.update({
                 'videoId': self._search_regex(r'videoId: [\'"](.+?)[\'"]', params, 'video id'),
                 'playerId': self._search_regex(r'playerId: [\'"](.+?)[\'"]', params, 'player id'),
                 'target': self._search_regex(r'target: [\'"](.+?)[\'"]', params, 'target'),
             })
-        else:
-            params = extract_attributes(self._search_regex(
-                r'(<[^>]+data-js="video-player"[^>]+>)',
-                webpage, 'player params element'))
+            return query
+
+        # No test for this case?
+        # xxx @remitamine in 4f427c4be860c582ca72dd4be64d45b54499232c
+        tag = self._search_regex(
+            r'(<[^>]+data-js="video-player"[^>]+>)',
+            webpage, 'player params element', default=None)
+        if tag:
+            params = extract_attributes(tag)
             query.update({
                 'videoId': params['data-video'],
                 'playerId': params['data-player'],
                 'target': params['id'],
             })
+            return query
+
+        iframe = self._search_regex(
+            r'<iframe[^>]+src="([^"]+)"', webpage, 'iframe player params')
+        query.update({
```

Comment:
i think it would be better to add support for this embed urls and return a url result instead of inline handling of the embed url.